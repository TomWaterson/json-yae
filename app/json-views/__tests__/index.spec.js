const { toHTML } = require("../index.js");

describe("Tree View", () => {
    it("toHTML should be defined", () => {
        // assert
        expect(toHTML).toBeDefined();
    });

    it("converts json to html", () => {
        let json = {
            hello: "world"
        };

        expect(toHTML(JSON.stringify(json))).not.toBe(null);
        expect(toHTML(JSON.stringify(json))).toBe(`<ul class="list-tree select-none"><li data-meta="string" class="item ml-6">
                <span class="property">hello</span>
                <span class="value">world</span>
            </li></ul>`);
    });
});
