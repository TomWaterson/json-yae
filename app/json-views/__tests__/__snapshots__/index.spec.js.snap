// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree View matches template 1`] = `"<ul class=\\"list-tree select-none\\">   <li data-meta=\\"object\\" class=\\"ml-2\\"> <svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 inline-block collapsible\\" viewBox=\\"0 0 20 20\\"><path d=\\"M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z\\"/></svg> <span class=\\"expander property\\">foo</span> <span class=\\"add\\"><svg class=\\"w-4 h-4 inline-block collapsible\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path d=\\"M11 9V5H9v4H5v2h4v4h2v-4h4V9h-4zm-1 11a10 10 0 1 1 0-20 10 10 0 0 1 0 20z\\"/></svg></span> <span class=\\"delete\\"><svg class=\\"w-4 h-4 inline-block collapsible\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path d=\\"M10 20a10 10 0 1 1 0-20 10 10 0 0 1 0 20zm5-11H5v2h10V9z\\"/></svg></span> <ul class=\\"nested\\">    <li data-meta=\\"string\\" class=\\"item ml-6\\"> <span class=\\"property\\">bar</span> <span class=\\"value\\">baz</span> </li>   </ul> </li>      <li data-meta=\\"string\\" class=\\"item ml-6\\"> <span class=\\"property\\">hello</span> <span class=\\"value\\">world</span> </li>   </ul>"`;
